app/services/degrees/map_from_apply.rb

```diff
@@ -47,47 +44,88 @@ def non_uk_degree_params
+    def qualification_type_params
+      qualification_type = find_dfe_reference_type
+
+      if qualification_type
+        { uk_degree: qualification_type[:name], uk_degree_uuid: qualification_type[:id] }
+      else
+        { uk_degree: attributes["qualification_type"] }
+      end
     end
 
-    def subject
-      course_subject = Dttp::CodeSets::DegreeSubjects::MAPPING.find do |k, _|
-        same_string?(k, attributes["subject"])
+    def institution_params
+      institution = find_dfe_reference_institution
+
+      if institution
+        { institution: institution[:name], institution_uuid: institution[:id] }
+      else
+        { institution: attributes["institution_details"] }
       end
+    end
 
-      course_subject.presence&.first || attributes["subject"]
+    def subject_params
+      subject = find_dfe_reference_subject
+
+      if subject
+        { subject: subject[:name], subject_uuid: subject[:id] }
+      else
+        { subject: attributes["subject"] }
+      end
     end
 
-    def qualification_type
-      degree_type = Dttp::CodeSets::DegreeTypes::MAPPING.find do |k, v|
-        same_hesa_code?(v[:hesa_code], attributes["hesa_degtype"]) ||
-        same_string?(v[:abbreviation], attributes["qualification_type"]) ||
-        same_string?(k, attributes["qualification_type"])
+    def grade_params
+      grade = find_dfe_reference_grade
+
+      if Dttp::CodeSets::Grades::MAPPING.keys.include?(grade[:name])
+        { grade: grade[:name], grade_uuid: grade[:id], other_grade: nil }
+      else
+        { grade: Dttp::CodeSets::Grades::OTHER, grade_uuid: grade[:id], other_grade: grade[:name] }
       end
+    end
 
-      degree_type.presence&.first || attributes["qualification_type"]
+    def uk_degree?
+      attributes["non_uk_qualification_type"].nil?
     end
 
-    def institution
-      institution_details = attributes["institution_details"].split(",").first
+    def find_dfe_reference_subject
+      find_dfe_reference_item(:subjects,
+                              uuid: attributes["degree_subject_uuid"],
+                              hesa_itt_code: sanitised_hesa(attributes["hesa_degsbj"]),
+                              name: attributes["subject"])
+    end
 
-      institution_type = Dttp::CodeSets::Institutions::MAPPING.find do |k, v|
-        same_hesa_code?(v[:hesa_code], attributes["hesa_degest"]) ||
-        same_string?(k, institution_details)
-      end
+    def find_dfe_reference_type
+      find_dfe_reference_item(:types,
+                              uuid: attributes["degree_type_uuid"],
+                              hesa_itt_code: sanitised_hesa(attributes["hesa_degtype"]),
+                              abbreviation: attributes["qualification_type"])
+    end
 
-      institution_type.presence&.first || institution_details
+    def find_dfe_reference_institution
+      find_dfe_reference_item(:institutions,
+                              uuid: attributes["degree_institution_uuid"],
```

Comment:
I've just got eyes on the actual naming now:

https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/presenters/concerns/qualification_api_data.rb#L46

```suggestion
                              uuid: attributes["institution_uuid"],
```